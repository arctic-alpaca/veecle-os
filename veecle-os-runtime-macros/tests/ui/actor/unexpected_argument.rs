#[derive(Debug, PartialEq, Clone, Default, veecle_os_runtime::Storable)]
pub struct Sensor(pub u8);

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(
    _sensor_reader: veecle_os_runtime::Reader<'_, Sensor>,
    _unexpected: u32,
) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(_unexpected: u32) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(_unexpected: <Foo as Bar>::Ty) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

#[veecle_os_runtime_macros::actor]
async fn macro_test_actor(_unexpected: [u32; 0]) -> std::convert::Infallible {
    unreachable!("We only care about the code compiling.");
}

fn main() {}
