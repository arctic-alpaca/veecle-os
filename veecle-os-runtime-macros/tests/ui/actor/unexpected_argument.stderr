error: only "Reader", "ExclusiveReader", "InitializedReader" and "Writer" arguments are allowed
 --> tests/ui/actor/unexpected_argument.rs:7:18
  |
7 |     _unexpected: u32,
  |                  ^^^

error: only "Reader", "ExclusiveReader", "InitializedReader" and "Writer" arguments are allowed
  --> tests/ui/actor/unexpected_argument.rs:13:40
   |
13 | async fn macro_test_actor(_unexpected: u32) -> std::convert::Infallible {
   |                                        ^^^

error: only "Reader", "ExclusiveReader", "InitializedReader" and "Writer" arguments are allowed
  --> tests/ui/actor/unexpected_argument.rs:18:40
   |
18 | async fn macro_test_actor(_unexpected: <Foo as Bar>::Ty) -> std::convert::Infallible {
   |                                        ^

error: only "Reader", "ExclusiveReader", "InitializedReader" and "Writer" arguments are allowed
  --> tests/ui/actor/unexpected_argument.rs:23:40
   |
23 | async fn macro_test_actor(_unexpected: [u32; 0]) -> std::convert::Infallible {
   |                                        ^^^^^^^^
